tests/buildkite/filter-pipeline.bats

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bats
+
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+function setup() {
+  src="tests/buildkite/pipeline-raw.yml"
+  filter_script=".buildkite/filter-pipeline.py"
+}
+
+@test "trigger step A" {
+  "$filter_script" --pipeline "$src" --git-diff "A" --output tests/buildkite/pipeline.yml
+  diff tests/buildkite/pipeline.yml tests/buildkite/golden/pipeline-A.yml
+}
+
+@test "trigger step C" {
+  "$filter_script" --pipeline "$src" --git-diff "C" --output tests/buildkite/pipeline.yml
+  diff tests/buildkite/pipeline.yml tests/buildkite/golden/pipeline-C.yml
+}
+
+@test "trigger step G" {
+  "$filter_script" --pipeline "$src" --git-diff "G" --output tests/buildkite/pipeline.yml
+  diff tests/buildkite/pipeline.yml tests/buildkite/golden/pipeline-G.yml
+}
+
```

Comment:
Nice tests, one thing that I think is missing tho (beside the one I mentioned in the previous comment) is to check what happens if one of the tests in the dependencies chain has no `changes_only` attribute.

Reply:
makes sense, I've updated the pipeline and added one more test